tests/DocumentationExamplesTest.php

```diff
@@ -1751,4 +1867,39 @@ private function assertInventoryCount($count): void
     {
         $this->assertCollectionCount($this->getDatabaseName() . '.' . $this->getCollectionName(), $count);
     }
+
+    private function waitForSnapshot(string $databaseName, string $collectionName): void
+    {
+        $collection = new Collection($this->manager, $databaseName, $collectionName);
+        $session = $this->manager->startSession(['snapshot' => true]);
+
+        // TODO: use hrtime() once the library requires PHP 7.3+
+        $startTime = microtime(true);
+
+        do {
+            try {
+                $collection->aggregate(
+                    [['$match' => ['_id' => ['$exists' => true]]]],
+                    ['session' => $session]
+                );
+
+                break;
+            } catch (CommandException $e) {
+                if ($e->getCode() === 246 /* SnapshotUnavailable */) {
+                    continue;
+                }
+
+                throw $e;
+            }
+        } while (microtime(true) < $startTime + 10);
```

Comment:
This is only called once, so I'm not inclined to parameterize the method; however, I do think this can be improved by removing the literal `10` here and changing the function to instead calculate the deadline up front (with a comment explaining the 10-second timeout).